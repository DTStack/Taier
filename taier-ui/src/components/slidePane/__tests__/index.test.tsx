import { cleanup, fireEvent, render } from '@testing-library/react';
import SlidePane from '..';

describe('Test SlidePane Component', () => {
    beforeEach(() => {
        cleanup();
    });

    it('Should match snapshot', () => {
        const { asFragment } = render(
            <SlidePane visible>
                <div>test</div>
            </SlidePane>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Should disabled when visible is false', () => {
        const { container } = render(
            <SlidePane visible={false}>
                <div>test</div>
            </SlidePane>
        );

        expect(container.querySelector<HTMLDivElement>('.dtc-slide-pane')?.style.pointerEvents).toBe('none');
    });

    it('Should support close pane by ESC', () => {
        const fn = jest.fn();
        const { container } = render(
            <SlidePane visible={false} onClose={fn}>
                <div>test</div>
            </SlidePane>
        );

        fireEvent.keyDown(container.querySelector('.dtc-slide-pane')!, { key: 'Escape' });

        expect(fn).toBeCalled();
    });
});
