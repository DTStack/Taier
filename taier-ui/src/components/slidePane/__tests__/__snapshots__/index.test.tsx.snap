// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SlidePane Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="dtc-slide-pane"
    style="top: 0px;"
    tabindex="-1"
  >
    <div
      class="dtc-slide-pane-conent"
      data-testid="slidepane_container"
      style="display: block; height: 100%;"
    >
      <div>
        test
      </div>
    </div>
    <span
      class="dtc-slide-pane-toggle"
      data-testid="slidepane_action"
    >
      <span
        aria-label="double-right"
        class="anticon anticon-double-right"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="double-right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
          />
        </svg>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
